### PR TITLE
Merge Galaxy-XSD into project

### PR DESCRIPTION
**Pull request by jmchilton**
Transplanted with commit authors maintained.

The most obvious advantages of it living here instead of its own project is that it gives it higher profile and it allows the version of the schema to match that of Galaxy (so we can check at least if the XML structure of a tool is valid for a given release). 

This has a few more advantages being in core that will probably come from follow up commits if this makes it in:
- Allow docs for tool XML generated from this to live in this project.
- Allow distribution with galaxy-lib.
- Allow checking testing tools as part of CI, ensures all new tool features are added to XSD and documented.

A Markdown document generated from this XSD is available for preview at https://github.com/jmchilton/planemo/blob/xsd_and_docs/docs/schema.md#tool%7Ctests%7Ctest%7Crepeat. I think in a subsequent commit I'll move those docs here - but for now it is a good way to "visualize" the XSD.

### CODE REVIEW

**Comment by martenson** [comment]
+1 on the principle of shipping xsd with Galaxy

**Comment by bgruening** [comment]
:+1:

**Comment by mvdbeek** [comment]
I think present and future tool authors will be very happy with this. 👍

File: lib/galaxy/tools/xsd/test.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#! /bin/bash
+if [ ! -d galaxy ];
+then
+    git clone https://github.com/galaxyproject/galaxy.git galaxy
+fi
+cd galaxy
+git pull
+cd ..
+
```
Comment (nsoranzo) [inline lib/galaxy/tools/xsd/test.sh]: I think that lines 2-9 can be removed.

File: lib/galaxy/tools/xsd/test.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+if [ ! -d tools-iuc ];
+then
+    git clone https://github.com/galaxyproject/tools-iuc.git tools-iuc
+fi
+cd tools-iuc
+git pull
+cd ..
+
+count=`wc -l tool_files.list | cut -f1 -d' '`
```
Comment (nsoranzo) [inline lib/galaxy/tools/xsd/test.sh]: There is no `tool_files.list` file.

**Review by nsoranzo — APPROVED** [review]

**Comment by jmchilton** [comment]
Pushed babaf7a2c675648e2851f81d985a72321b4eeb44 to address reviewer comments on XSD.
- Merged in @bgruening's numerous awesome fixes from Galaxy-XSD downstream.
- Removed test script with issues that make if not a fit for this repo in that form (noted by @nsoranzo).
- Incorporate other changes to the XSD suggested by @nsoranzo.

File: lib/galaxy/tools/xsd/LICENSE [hunk 3]
```diff
@@ -0,0 +1,22 @@
+The MIT License (MIT)
+
+Copyright (c) 2011-2015 John Chilton, Jean-Frédéric Berthelot, Pierre Lindenbaum
```
Comment (bgruening) [inline lib/galaxy/tools/xsd/LICENSE]: 2016

**Comment by bgruening** [comment]
Thanks @jmchilton! Ready for merge?

File: lib/galaxy/tools/xsd/LICENSE (hunk 3 above)
Comment (jmchilton) [reply lib/galaxy/tools/xsd/LICENSE]: Updated.

**Comment by jmchilton** [comment]
I consider it ready for merging. I'll open a new PR regarding the LICENSE (let's see if we can drop it as a special license inside of Galaxy) - though I did update the year.

**Comment by martenson** [comment]
I would leave it open for a bit - it deserves to be open for more than 14 hours I believe.

**Comment by jmchilton** [comment]
I'm pretty sure we can relocate this here and keep the LICENSE statement as it is, but it would be cleaner if we re-licensed it under Galaxy's license.

@JeanFred, @lindenb as co-copyright holders I was hoping to get your explicit permission to re-license this under the terms of [Galaxy's license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt). It is a permissive (not copy-left) license very much in the spirit of the original MIT license I believe. Regardless, this PR adds both of your names to Galaxy's CONTRIBUTORS list.

xref https://github.com/galaxyproject/Galaxy-XSD/issues/2

**Review by martenson — APPROVED** [review]

**Comment by JeanFred** [comment]
Wow, I’m really, really happy to see this integrated into core Galaxy!

> @JeanFred, @lindenb as co-copyright holders I was hoping to get your explicit permission to re-license this under the terms of Galaxy's license. It is a permissive (not copy-left) license very much in the spirit of the original MIT license I believe. Regardless, this PR adds both of your names to Galaxy's CONTRIBUTORS list.

Sure, I am fine with re-licensing or dual-licensing my contributions to make it easier to manage for the Galaxy project. I was not very familiar with AFL but it looks good to me. To make it official sounding:

> I hereby agree to license my contributions under the Academic Free License version 3.0 

Thanks everyone!

**Pull request merged**